modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/PlacementDriverManager.java

```diff
@@ -187,10 +335,264 @@ private void takeOverActiveActor() {
      */
     private void stepDownActiveActor() {
         isActiveActor = false;
+
+        stopUpdater();
     }
 
     @TestOnly
     boolean isActiveActor() {
         return isActiveActor;
     }
+
+    /**
+     * Starts a dedicated thread to renew or assign leases.
+     */
+    private void startUpdater() {
+        //TODO: IGNITE-18879 Implement lease maintenance.
+        updater = new Thread(NamedThreadFactory.threadPrefix(clusterService.topologyService().localMember().name(), "lease-updater")) {
+            @Override
+            public void run() {
+                while (isActiveActor) {
+                    for (Map.Entry<ReplicationGroupId, Set<Assignment>> entry : groupAssignments.entrySet()) {
+                        ReplicationGroupId grpId = entry.getKey();
+
+                        ReplicationGroupLease lease = replicationGroups.getOrDefault(grpId, new ReplicationGroupLease());
+
+                        HybridTimestamp now = clock.now();
+
+                        if (lease.stopLeas == null || now.getPhysical() > (lease.stopLeas.getPhysical() - LEASE_PERIOD / 2)) {
+                            var leaseKey = ByteArray.fromString(PLACEMENTDRIVER_PREFIX + grpId);
+
+                            var newTs = new HybridTimestamp(now.getPhysical() + LEASE_PERIOD, 0);
+
+                            ClusterNode holder = nextLeaseHolder(entry.getValue());
+
+                            if (lease.leaseholder == null && holder != null
+                                    || lease.leaseholder != null && lease.leaseholder.equals(holder)
+                                    || holder != null && (lease.stopLeas == null || now.getPhysical() > lease.stopLeas.getPhysical())) {
```

Comment:
does this mean the meta storage invoke happens only if now is greater than lease expiration time, how it combines with condition above `lease.stopLeas.getPhysical() - LEASE_PERIOD / 2` ?

Reply:
We may not waiting to lease expiration in case whrn the leaseholder in not changed, but if the leaseholder is changed the thread will wait to lease expiration.